### PR TITLE
Add type tokens to storybook

### PR DESCRIPTION
**Pull request by jameskoster**
Screenshot 2024-10-09 at 14 49 42

Related to https://github.com/WordPress/gutenberg/pull/65982.

## What?
Document the recently added type tokens (https://github.com/WordPress/gutenberg/pull/65418) in Storybook.

## Why?
This makes them more discoverable and usable.

## Testing Instructions
1. `npm run storybook`
2. Browse the Tokens -> Typography section

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Looks great. I'll defer to a review from the components team if they have one.

Storybook arcana-wise, it's interesting to use HTML tables here, glad to see that working. I looked at using Markdown tables, that seems to [require an extension for the moment](https://storybook.js.org/docs/writing-docs/mdx#markdown-tables-arent-rendering-correctly).

File: storybook/stories/tokens/typography.mdx
Comment (jasmussen) [inline storybook/stories/tokens/typography.mdx]: Should it be "SASS"?

**Comment by tyxla** [comment]
Will take a look, but in the meantime, I've converted spaces to tabs to be consistent with the rest of the codebase.

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [reply storybook/stories/tokens/typography.mdx]:
```suggestion
They are defined as SASS mixins and can be used like so:
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
Primitive tokens are organized into 4 sets relating to `size`, `line-height`, `weight`, and `family`, defined as SASS variables.
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]: We can get some syntax highlighting if we specifically mark this code block as a CSS one.

```suggestion
```css
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
```css
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
```css
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
```css
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
```css
```

File: storybook/stories/tokens/typography.mdx
Comment (tyxla) [inline storybook/stories/tokens/typography.mdx]:
```suggestion
```css
```

**Review by tyxla — APPROVED** [review]
Looks great! We can always add more context or explanations if we feel it's needed. 🚀 

After addressing spacing, I also left a few minor suggestions, mostly related to syntax highlighting. Good to go after they're addressed. 👍

File: storybook/stories/tokens/typography.mdx
Comment (ciampo) [reply storybook/stories/tokens/typography.mdx]: Since this is SASS, should try this instead?

```suggestion
```sass
```
or maybe 

```suggestion
```scss
```

Not sure if either of them is supported

Comment (ciampo) [reply storybook/stories/tokens/typography.mdx]: The same would apply to other SASS code snippets

Comment (tyxla) [reply storybook/stories/tokens/typography.mdx]: None of those 2 are supported (I tried them out while reviewing the PR), which is why I suggested `css`.

Comment (ciampo) [reply storybook/stories/tokens/typography.mdx]: Thank you for clarifying!

**Comment by jameskoster** [comment]
Thanks the for the reviews y'all.

> I looked at using Markdown tables, that seems to [require an extension for the moment](https://storybook.js.org/docs/writing-docs/mdx#markdown-tables-arent-rendering-correctly).

Yeah I meant to note that. Markdown tables aren't working currently. Let's look at adding the extension separately then revisit.

**Pull request merged**
